evmjit/libevmjit/Arith256.cpp

```diff
@@ -343,18 +408,49 @@ llvm::Function* Arith256::getMulModFunc()
 
 llvm::Value* Arith256::mul(llvm::Value* _arg1, llvm::Value* _arg2)
 {
+	if (auto c1 = llvm::dyn_cast<llvm::ConstantInt>(_arg1))
+	{
+		if (auto c2 = llvm::dyn_cast<llvm::ConstantInt>(_arg2))
+			return Constant::get(c1->getValue() * c2->getValue());
+	}
+
 	return createCall(getMulFunc(), {_arg1, _arg2});
 }
 
 std::pair<llvm::Value*, llvm::Value*> Arith256::div(llvm::Value* _arg1, llvm::Value* _arg2)
 {
-	auto div =  m_builder.CreateExtractValue(createCall(getDivFunc(Type::Word), {_arg1, _arg2}), 0, "div");
-	auto mod =  m_builder.CreateExtractValue(createCall(getDivFunc(Type::Word), {_arg1, _arg2}), 1, "mod");
+	if (auto c1 = llvm::dyn_cast<llvm::ConstantInt>(_arg1))
+	{
```

Comment:
Not sure... The rule is `c. Don't use braces for condition-body one-liners.`, not one-statements... I think it's more readable, but you can decide.